### PR TITLE
Create Docker image for ART

### LINKED ISSUE #370: Streamlining ART setup process
**Is your feature request related to a problem? Please describe.**
The current ART setup and development environment could benefit some standardization and clarification for both new comers and existing contributors.  

* For instance, as it stands, many contributors are using different versions of the pip packages requirements  which can lead to confusion and time wasted when debugging potentially platform specific bugs. 
* Also some packages (such as xgboost) seem to be causing problems on some platforms (ie: MacOS) but not on others.
* Readme documentation could also be improved to document various conventions used within the code base (such as where the datasets are downloaded)  and conventions used within ART tests

**Describe the solution you'd like**
1. A dockerised version of ART which would provide a standardized running environment across platforms when dealing platform specific bugs
2. Separating the dataset downloads from the tests runs and/or mentioning to new users that various datasets should be downloaded prior to running the tests AND potentially changing/removing the use of `Config.jsons`
3. removing duplicate packages from  `requirements.txt` and `test_requirements.txt` files
4. making sure `requirement` packages all have a specified version associated with them (to guarantee a specific combination of packages does pass tests on travis)
5. letting users install only their framework of choice (and not all of them) in order to use or test the ART framework

### PR DESCRIPTION
**Pull request by killianlevacher**
# Description

Tried to address as many setup issues as possible to make things clearer for new comers. Also Dockerised the ART project to make sure we have a consistent reproducible environment across platforms. Updated README documentation accordingly.

Fixes #370

## Type of change

Please check all relevant options.

- [ ] Bug fix (non-breaking)
- [X] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: CONTRIBUTING-cn.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+＃促進對抗性魯棒性工具箱
```
Comment (beat-buesser) [inline CONTRIBUTING-cn.md:1]: Great idea!

File: README-cn.md [hunk 2]
```diff
@@ -142,20 +160,14 @@ python examples/<example_name>.py
...
-```bash
-git commit -s -m 'Add new feature'
-```
+看到 [CONTRIBUTING-cn.md](CONTRIBUTING-cn.md)
 
 ## 引用ART
 如果您使用ART进行研究，请考虑引用以下参考文件：
 ```
-@article{art2018,
+@article{art2018,~~~~
```
Comment (beat-buesser) [inline README-cn.md:170]: Do we need these 4 tilde characters?

Comment (killianlevacher) [reply README-cn.md:170]: no I don't think we do. I didn't realise that was inserted. I just removed them

File: test_requirements.txt [hunk 3]
```diff
@@ -1,22 +1,23 @@
...
+keras
 tensorflow_addons
-scikit-learn==0.22.1
-xgboost==1.0.0
-lightgbm==2.3.1
+mxnet>=1.6.0
+xgboost>=1.0.0
+lightgbm>=2.3.1
+#torch>=1.2.0
+torch>=1.4.0
```
Comment (beat-buesser) [inline test_requirements.txt:16]:
```suggestion
torch==1.3.1
```

**Pull request merged**